WooCommerce/Classes/POS/Card Present Payments/CardPresentPaymentOnboardingAdaptor.swift

```diff
@@ -31,10 +31,6 @@ final class CardPresentPaymentsOnboardingPresenterAdaptor: CardPresentPaymentsOn
     ///   - completion: Callback when the onboarding is complete
     func showOnboardingIfRequired(from viewController: ViewControllerPresenting,
                                   readyToCollectPayment completion: @escaping () -> Void) {
-        guard readinessSubscription == nil else {
```

Comment:
Yes, I don't know about this as well 🤔 maybe @joshheald you have a better idea? I assume it was designed to prevent show more than one onboarding at the same time.

@jaclync would setting it `nil` here help prevent issues when onboarding is not completed but we dismiss the onboarding view?
```
    private func showOnboarding(from viewController: ViewControllerPresenting,
                                readyToCollectPayment completion: @escaping () -> Void) {
        let onboardingViewController = CardPresentPaymentsOnboardingViewController(viewModel: onboardingViewModel,
                                                                                   onWillDisappear: { [weak self] in
            self?.readinessSubscription?.cancel() 
           self?.readinessSubscription = nil <----
        })
```

Reply:
> Yes, I don't know about this as well 🤔 maybe @joshheald you have a better idea? I assume it was designed to prevent show more than one onboarding at the same time.
> 
> @jaclync would setting it `nil` here help prevent issues when onboarding is not completed but we dismiss the onboarding view?
> 
> ```
>     private func showOnboarding(from viewController: ViewControllerPresenting,
>                                 readyToCollectPayment completion: @escaping () -> Void) {
>         let onboardingViewController = CardPresentPaymentsOnboardingViewController(viewModel: onboardingViewModel,
>                                                                                    onWillDisappear: { [weak self] in
>             self?.readinessSubscription?.cancel() 
>            self?.readinessSubscription = nil <----
>         })
> ```

Great suggestion on cancelling/nilling the subscription instead -

This seems to be in the [UIKit version](https://github.com/woocommerce/woocommerce-ios/blob/316063cc980ebf2231fa69411940364f3e1706ec/WooCommerce/Classes/ViewRelated/Orders/Collect%20Payments/CardPresentPaymentsOnboardingPresenter.swift#L44-L49) of the onboarding presenter, where it creates the onboarding VC which includes a closure when the view will disappear to cancel the subscription.

In the new adapted version of the onboarding presenter, a view model is emitted instead of a view where we can detect its onDisappear. After a deeper look, I reverted this removal of nil check and use a closure in the `showOnboarding` event to nil out the subscription on view's `onDisappear` in https://github.com/woocommerce/woocommerce-ios/pull/14185/commits/dbdc9a77425b7aab5a4513a9838855e7da9904ab. Lemme know what you think!